demos/common/samples/args_helper.hpp

```diff
@@ -85,3 +85,54 @@ void parseInputFilesArguments(std::vector<std::string> &files) {
         readInputFilesArguments(files, args.at(i));
     }
 }
+
+std::vector<std::string> split(const std::string &s, char delim) {
+    std::vector<std::string> result;
+    std::stringstream ss(s);
+    std::string item;
+
+    while (getline(ss, item, delim)) {
+        result.push_back(item);
+    }
+    return result;
+}
+
+std::vector<std::string> parseDevices(const std::string& device_string) {
+    std::string comma_separated_devices = device_string;
+    if (comma_separated_devices.find(":") != std::string::npos) {
+        comma_separated_devices = comma_separated_devices.substr(comma_separated_devices.find(":") + 1);
+    }
+    auto devices = split(comma_separated_devices, ',');
+    for (auto& device : devices)
+        device = device.substr(0, device.find("("));
+    return devices;
+}
+
+std::map<std::string, uint32_t> parseValuePerDevice(const std::set<std::string>& devices,
+                                                    const std::string& values_string) {
+    //  Format: <device1>:<value1>,<device2>:<value2> or just <value>
+    auto values_string_upper = values_string;
+    std::transform(values_string_upper.begin(),
+                   values_string_upper.end(),
+                   values_string_upper.begin(),
+                   [](unsigned char c){ return std::toupper(c); });
+    std::map<std::string, uint32_t> result;
+    auto device_value_strings = split(values_string_upper, ',');
+    for (auto& device_value_string : device_value_strings) {
+        auto device_value_vec =  split(device_value_string, ':');
+        if (device_value_vec.size() == 2) {
+            auto it = std::find(devices.begin(), devices.end(), device_value_vec.at(0));
+            if (it != devices.end()) {
+                result[device_value_vec.at(0)] = std::stoi(device_value_vec.at(1));
+            }
```

Comment:
IMO, there should be an `else` to this `if` that throws an exception. If the condition in the `if` is false, then the user is trying to set a value for a device that isn't used.

Reply:
This will require to keep `values_string` consistent with used `devices`, which will be annoying for searching the best parameters for a demo manually.

Reply:
OK, fair enough.